test/getRandomString.test.ts

```diff
@@ -0,0 +1,43 @@
+import { getRandomString } from '../src/index' // Update this import path to match your setup
+import { expect, it, describe } from '@jest/globals'
+
+describe('getRandomString', function () {
+    // Basic Test
+    it('should produce a string of the desired length', () => {
+        const length = 18
+        const result = getRandomString(length)
+        expect(result).toHaveLength(length)
+    })
+
+    // Charset Test
+    it('should produce a string containing only valid characters', () => {
+        const charset = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789'
+        const result = getRandomString(1000) // test a long string to get a variety of characters
+        for (const char of result) {
+            expect(charset).toContain(char)
```

Comment:
Low level optimisation if this ever becomes a speed bottleneck - we can do a range check as the ASCII layout is contiguous which results in O(1) rather than O(n^2) traversal

Reply:
should be fine for now, these wont be longer than 20-30 chars

Reply:
Yep this isn't a change worth doing, more of a musing